pkg/stdlib/html/blob.go

```
@@ -0,0 +1,145 @@
+ * @param params (Object) - Optional, An object containing the following properties :
+ * 		x (Float|Int) - Optional, X position of the viewport.
+ * 		x (Float|Int) - Optional,Y position of the viewport.
+ * 		width (Float|Int) - Optional, Width of the viewport.
+ * 		height (Float|Int) - Optional, Height of the viewport.
+ * 		format (String) - Optional, Either "jpeg" or "png".
+ * 		quality (Int) - Optional, Quality, in [0, 100], only for jpeg format.
+ * @returns data (Binary) - Returns a base64 encoded string in binary format.
+ */
+func Screenshot(ctx context.Context, args ...core.Value) (core.Value, error) {
+	err := core.ValidateArgs(args, 1, 2)
+	if err != nil {
+		return values.None, err
+	}
+
+	arg1 := args[0]
+
+	err = core.ValidateType(arg1, core.HTMLDocumentType, core.StringType)
+	if err != nil {
+		return values.None, err
+	}
+
+	var doc *dynamic.HTMLDocument
+	var ok bool
+	if arg1.Type() == core.StringType {
+		buf, err := Document(ctx, arg1, values.NewBoolean(true))
+		if err != nil {
+			return values.None, err
+		}
+		doc, ok = buf.(*dynamic.HTMLDocument)
+		defer doc.Close()
+	} else {
+		doc, ok = arg1.(*dynamic.HTMLDocument)
+	}
+
+	if !ok {
+		return values.None, core.Error(core.ErrInvalidType, "expected dynamic document")
+	}
+
+	screenshotParams := &dynamic.ScreenshotArgs{
+		X:       0,
+		Y:       0,
+		Width:   -1,
+		Height:  -1,
+		Format:  "jpeg",
+		Quality: 100,
+	}
+	if len(args) == 2 {
+		arg2 := args[1]
+		err = core.ValidateType(arg2, core.ObjectType)
+		if err != nil {
+			return values.None, err
+		}
+		params, ok := arg2.(*values.Object)
+		if !ok {
+			return values.None, core.Error(core.ErrInvalidType, "expected object")
+		}
+
+		format, found := params.Get("format")
+		if found {
+			err = core.ValidateType(format, core.StringType)
+			if err != nil {
+				return values.None, err
+			}
+			if !dynamic.IsScreenshotFormatValid(format.String()) {
+				screenshotParams.Format = dynamic.ScreenshotFormatJPEG
```

Basically, instead of the fallback value, let's return an error.